homeassistant/components/media_player/anthemav.py

```diff
@@ -0,0 +1,225 @@
+        self.hass = hass
+        self._host = host
+        self._port = port
+        self.avr = None
+
+    def _poll_and_return(self, propname, dval):
+        if self.reader:
+            pval = getattr(self.reader, propname)
+            _LOGGER.debug('query '+propname+' returned from avr: '+str(pval))
+            return pval
+        else:
+            _LOGGER.debug('query '+propname+' returned default: '+str(dval))
+            return dval
+
+    def _poll_or_null(self, propname):
+        if self.reader:
+            pval = getattr(self.reader, propname)
+            _LOGGER.debug('query '+propname+' returned from avr: '+str(pval))
+            return pval
+        else:
+            return
+
+    @property
+    def reader(self):
+        """Expose the protocol with smart wrapper."""
+        if hasattr(self, 'avr'):
+            if hasattr(self.avr, 'protocol'):
+                return self.avr.protocol
+
+    @reader.setter
+    def reader(self, value):
+        self.avr.protocol = value
+
+    @property
+    def supported_media_commands(self):
+        """Return flag of media commands that are supported."""
+        return SUPPORT_ANTHEMAV
+
+    @property
+    def name(self):
+        """Return name of device."""
+        return self._poll_and_return('model', DEFAULT_NAME)
+
+    @property
+    def state(self):
+        """Return state of power on/off."""
+        pwrstate = self._poll_or_null('power')
+
+        if pwrstate is True:
+            return STATE_ON
+        elif pwrstate is False:
+            return STATE_OFF
+        else:
+            return STATE_UNKNOWN
+
+    @property
+    def volume_level(self):
+        """Return volume level from 0 to 1."""
+        return self._poll_and_return('volume_as_percentage', 0.0)
+
+    @property
+    def media_title(self):
+        """Return current input name (closest we have to media title)."""
+        return self._poll_and_return('input_name', 'No Source')
+
+    @property
+    def app_name(self):
+        """Return details about current video and audio stream."""
+        return self._poll_and_return('video_input_resolution_text', '') + ' ' \
+            + self._poll_and_return('audio_input_name', '')
+
+    @property
+    def source(self):
+        """Return currently selected input."""
+        return self._poll_and_return('input_name', "Unknown")
+
+    @property
+    def source_list(self):
+        """Return all active, configured inputs."""
+        return self._poll_and_return('input_list', ["Unknown"])
+
+    def media_play(self):
+        """Unsupported."""
+        return
+
+    def select_source(self, source):
+        """Change AVR to the designated source (by name)."""
+        self.update_avr('input_name', source)
+
+    def turn_off(self):
+        """Turn AVR power off."""
+        self.update_avr('power', False)
+
+    def turn_on(self):
+        """Turn AVR power on."""
+        self.update_avr('power', True)
+
+    def volume_up(self):
```

Comment:
Delete `volume_up` and `volume_down`. They are implemented by `MediaPlayerDevice` if there isn't a platform specific implementation.